### PR TITLE
[12.0] [ADD] purchase_supplier_free_shipping_threshold

### PR DESCRIPTION
**Pull request by victor-champonnois**
https://gestion.coopiteasy.be/mail/view?model=project.task&res_id=12658

### CODE REVIEW

**Review by huguesdk — CHANGES_REQUESTED** [review]
looks good to me, except for the name of the field and the module. “franco” doesn’t seem to mean anything in english. i would advise to use “free_shipping_threshold”.

moreover, did you take a look at the `delivery.carrier.free_over` field in the standard `delivery` module?

**Comment by victor-champonnois** [comment]
@huguesdk yes good idea to rename free_shipping_threshold. I fixed that.
Yes I know about the delivery module, but it's for delivering to clients, here the threshold is for the delivery costs at reception.

File: purchase_supplier_free_shipping_threshold/models/purchase_order.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# - Vincent Van Rossem <vincent@coopiteasy.be>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    free_shipping_threshold = fields.Monetary(related="partner_id.franco")
```
Comment (huguesdk) [inline purchase_supplier_free_shipping_threshold/models/purchase_order.py]:
```suggestion
    free_shipping_threshold = fields.Monetary(related="partner_id.free_shipping_threshold")
```

**Review by huguesdk — CHANGES_REQUESTED** [review]
still one `franco` to replace. 😉 also, to make pre-commit pass, you need to add the `setup/purchase_supplier_free_shipping_threshold` directory.

**Review by huguesdk — APPROVED** [review]
all good now!

**Comment by victor-champonnois** [comment]
@huguesdk a thousand thanks :pray:

**Comment by victor-champonnois** [comment]
@huguesdk I squashed my commits.

**Pull request merged**